### PR TITLE
fix: `noAriaHiddenonFocusable` when using a JSX expression

### PR DESCRIPTION
**Pull request by anthonyshew**
## Summary

I ran into a bug with JSX expressions for this rule, as seen in https://github.com/biomejs/biome/issues/3689. In this PR, we're handling the various ways you can define this attribute with slightly more robustness.

## Test Plan

I've added to the `valid.jsx` file for this rule to recreate the bug I ran into. It passes now!

### CODE REVIEW

File: crates/biome_js_analyze/src/lib.rs [hunk 1]
```diff
@@ -204,7 +203,7 @@ mod tests {
             dependencies_index: Some(1),
             stable_result: StableHookResult::None,
         };
-        let rule_filter = RuleFilter::Rule("style", "useNodejsImportProtocol");
+        let rule_filter = RuleFilter::Rule("a11y", "noAriaHiddenOnFocusable");
```
Comment (dyc3) [inline crates/biome_js_analyze/src/lib.rs:206]: You'll need to revert the changes to the quick test.

File: crates/biome_js_analyze/src/lint/a11y/no_aria_hidden_on_focusable.rs [hunk 2]
```diff
@@ -52,6 +59,40 @@ declare_lint_rule! {
     }
 }
 
+declare_node_union! {
```
Comment (anthonyshew) [inline crates/biome_js_analyze/src/lint/a11y/no_aria_hidden_on_focusable.rs:62]: This implementation is copy-pasted from https://github.com/biomejs/biome/blob/8d29726acb4596b5abcfae1ddb48cdf2d1bcbe86/crates/biome_js_analyze/src/lint/a11y/no_noninteractive_tabindex.rs#L61.

I'm not certain if that warrants breaking it out into some shared module or if refactoring in a separate PR is more preferable. Happy to do what maintainers think is best!

**Review by dyc3 — DISMISSED** [review]

File: crates/biome_js_analyze/src/lib.rs (hunk 1 above)
Comment (anthonyshew) [reply crates/biome_js_analyze/src/lib.rs]: Already done! 👍

File: crates/biome_js_analyze/src/lint/a11y/no_aria_hidden_on_focusable.rs (hunk 2 above)
Comment (ematipico) [reply crates/biome_js_analyze/src/lint/a11y/no_aria_hidden_on_focusable.rs:62]: We usually move shared code whenever it's possible. For your case, you'll have to update `biome_js_syntax/src/expr_ext.rs`, copy-paste the new node and expose it publicly.

**Review by ematipico — APPROVED** [review]
Thank you very much for the bug fix. Everything looks good. I would complete the PR with:
- making the code shared, now that there's a use case
- [add a changelog line to `CHANGELOG.md`](https://github.com/biomejs/biome/blob/main/CONTRIBUTING.md#changelog)

File: crates/biome_js_analyze/src/lint/a11y/no_aria_hidden_on_focusable.rs (hunk 2 above)
Comment (anthonyshew) [reply crates/biome_js_analyze/src/lint/a11y/no_aria_hidden_on_focusable.rs]: Sounds great! Done!
